vm/src/stdlib/re.rs

```diff
@@ -237,16 +237,14 @@ fn do_split(
     // essentially Regex::split, but it outputs captures as well
     let mut output = Vec::new();
     let mut last = 0;
-    let mut n = 0;
-    for captures in pattern.regex.captures_iter(text) {
+    for (n, captures) in pattern.regex.captures_iter(text).enumerate() {
         let full = captures.get(0).unwrap();
         let matched = &text[last..full.start()];
         last = full.end();
         output.push(Some(matched));
         for m in captures.iter().skip(1) {
             output.push(m.map(|m| m.as_bytes()));
         }
-        n += 1;
         if maxsplit != 0 && n >= maxsplit {
```

Comment:
is this equivalent to before? For the first loop, `n` was 1 in the original code but now 0 in new code.

Reply:
I think he did have n+1 >= maxsplit originally, so maybe that didn't actually work? @skinny121